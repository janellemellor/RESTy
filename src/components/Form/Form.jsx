import React from 'react';
import PropTypes from 'prop-types';

const Form = ({ onSubmit, url, onUrlChange, method, onInputChange, jsonBody, onJsonBodyChange }) => (
  <form onSubmit={onSubmit}>
    <fieldset>
      <input name="url-input" type="text" value={url} onUrlChange={onUrlChange} />
      <label> GET
        <input type="radio" name="methods" value="GET" onInputChange={onInputChange}/>
      </label>
      <label> POST
        <input type="radio" name="methods" value="POST" onInputChange={onInputChange}/>
      </label>  
      <label> PUT
        <input type="radio" name="methods" value="PUT" onInputChange={onInputChange}/>
      </label>  
      <label> PATCH
        <input type="radio" name="methods" value="PATCH" onInputChange={onInputChange}/>
      </label>  
      <label> DELETE
        <input type="radio" name="methods" value="DELETE" onInputChange={onInputChange}/>
      </label>    
      <button>Go!</button>
      <label>
        <textarea name="jsonBody" value={jsonBody} onJsonBodyChange={onJsonBodyChange}> </textarea>
      </label>
    </fieldset>
  </form>
);

Form.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  url: PropTypes.string.isRequired,
  onUrlChange: PropTypes.func.isRequired,
  method: PropTypes.string.isRequired,
  onInputChange: PropTypes.func.isRequired,
  jsonBody: PropTypes.string.isRequired,
  onJsonBodyChange: PropTypes.func.isRequired
};

export default Form;


