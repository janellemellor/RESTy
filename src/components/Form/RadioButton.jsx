import React from 'react';
import PropTypes from 'prop-types';

export const RadioButtonGroup = ({ children, name, onChange, method }) => {
  const radioButtons = React.Children.map(children, child => {
    return React.cloneElement(child, {
      name,
      onChange,
      method 
    });
  });

  return (
    <>
      {radioButtons}
    </>
  );
};

export const RadioButton = ({ method, name, value, onChange }) => (
  <>
    <input type="radio" checked={method === value} name={name} value={value} onChange={onChange} />
    <label htmlFor={value}>{value}</label>
  </>
);


RadioButtonGroup.propTypes = {
  children: PropTypes.node,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  method: PropTypes.string.isRequired
};

RadioButton.propTypes = {
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
  method: PropTypes.string.isRequired
};

