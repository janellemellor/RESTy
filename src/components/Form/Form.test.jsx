import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form.jsx';

describe('Form Component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Form 
      onSubmit={() => {}}
      url='https://github.com/janellemellor'
      onChange={() => {}}
      jsonBody='{}' />);
    expect(wrapper).toMatchSnapshot();
  });    
});
