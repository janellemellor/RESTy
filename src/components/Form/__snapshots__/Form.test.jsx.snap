// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component matches a snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <fieldset>
    <input
      name="url-input"
      onUrlChange={[Function]}
      type="text"
      value="https://github.com/janellemellor"
    />
    <label>
       GET
      <input
        name="methods"
        onInputChange={[Function]}
        type="radio"
        value="get"
      />
    </label>
    <label>
       POST
      <input
        name="methods"
        onInputChange={[Function]}
        type="radio"
        value="get"
      />
    </label>
    <label>
       PUT
      <input
        name="methods"
        onInputChange={[Function]}
        type="radio"
        value="get"
      />
    </label>
    <label>
       PATCH
      <input
        name="methods"
        onInputChange={[Function]}
        type="radio"
        value="get"
      />
    </label>
    <label>
       DELETE
      <input
        name="methods"
        onInputChange={[Function]}
        type="radio"
        value="get"
      />
    </label>
    <button>
      Go!
    </button>
    <label>
      <textarea
        name="jsonBody"
      >
         
      </textarea>
    </label>
  </fieldset>
</form>
`;
